spark/src/main/scala/org/apache/spark/sql/delta/util/FileNames.scala

```diff
@@ -18,24 +18,34 @@ package org.apache.spark.sql.delta.util
 
 import java.util.UUID
 
+import org.apache.spark.sql.delta.DeltaLog
 import org.apache.hadoop.fs.{FileStatus, Path}
 
 /** Helper for creating file names for specific commits / checkpoints. */
 object FileNames {
 
   val deltaFileRegex = raw"(\d+)\.json".r
+  val uuidDeltaFileRegex = raw"(\d+)\.[^.]+\.json".r
   val compactedDeltaFileRegex = raw"(\d+).(\d+).compacted.json".r
   val checksumFileRegex = raw"(\d+)\.crc".r
   val checkpointFileRegex = raw"(\d+)\.checkpoint((\.\d+\.\d+)?\.parquet|\.[^.]+\.(json|parquet))".r
 
-  val deltaFilePattern = deltaFileRegex.pattern
-  val compactedDeltaFilePattern = compactedDeltaFileRegex.pattern
-  val checksumFilePattern = checksumFileRegex.pattern
-  val checkpointFilePattern = checkpointFileRegex.pattern
+  private val deltaFilePattern = deltaFileRegex.pattern
+  private val uuidDeltaFilePattern = deltaFileRegex.pattern
+  private val compactedDeltaFilePattern = compactedDeltaFileRegex.pattern
+  private val checksumFilePattern = checksumFileRegex.pattern
+  private val checkpointFilePattern = checkpointFileRegex.pattern
 
   /** Returns the delta (json format) path for a given delta file. */
   def deltaFile(path: Path, version: Long): Path = new Path(path, f"$version%020d.json")
 
+  /** Returns the delta (json format) path for a given delta file. */
+  def uuidDeltaFile(logPath: Path, version: Long): Path = {
+    val basePath = commitDirPath(logPath)
+    val uuid = UUID.randomUUID.toString
```

Comment:
```suggestion
    val uuid = UUID.randomUUID
```
(`$uuid` will `toString` automatically)